sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -204,6 +204,29 @@ case class Elt(children: Seq[Expression])
       }
     }
   }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val index = children.head.map(_.genCode(ctx))(0)
+    val strings = children.tail.map(_.genCode(ctx))
+    val stringValues = strings.map { eval =>
+      s"${eval.isNull} ? null : ${eval.value}"
+    }.mkString(", ")
+    val indexVal = ctx.freshName("index")
+    val stringArray = ctx.freshName("strings");
+
+    ev.copy(index.code + "\n" + strings.map(_.code).mkString("\n") + s"""
+      int $indexVal = ${index.value} - 1;
+      UTF8String[] $stringArray = {$stringValues};
```

Comment:
After re-checking the wholestage code, I find that is not correct. If any expression in a sub plan will fall back, then the sub plan will not be wholestage codegen. The whole plan can be wholestage codegen, but for these no wholestage codegen sub plans, they will be wrapped with input adapter.